src/Microsoft.OData.Core/UriParser/SemanticAst/KeySegment.cs

```diff
@@ -45,14 +45,30 @@ public sealed class KeySegment : ODataPathSegment
         /// <exception cref="ODataException">Throws if the input entity set is not related to the input type.</exception>
         [SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures", Justification = "Using key value pair is exactly what we want here.")]
         public KeySegment(IEnumerable<KeyValuePair<string, object>> keys, IEdmEntityType edmType, IEdmNavigationSource navigationSource)
+            : this(keys.ToList(), edmType, navigationSource, verifyNavigationSourceType: true)
         {
-            this.keys = new ReadOnlyCollection<KeyValuePair<string, object>>(keys.ToList());
+        }
+
+        /// <summary>
+        /// Construct a segment that represents a key lookup.

```

Comment:
```suggestion
        /// Constructs a segment that represents a key lookup.
```

Reply:
Same term used on the other constructors. But I've updated this as well as the docs of the other constructors in this class.